milli/src/search/facet/filter.rs

```diff
@@ -100,10 +100,10 @@ impl<'a> Filter<'a> {
                         }
                     }
 
-                    if ors.len() > 1 {
-                        ands.push(FilterCondition::Or(ors));
-                    } else if ors.len() == 1 {
-                        ands.push(ors.pop().unwrap());
+                    match ors.len() {
+                        1 => ands.push(ors.pop().unwrap()),
+                        n if n > 1 => ands.push(FilterCondition::Or(ors)),
+                        _ => (),
```

Comment:
```suggestion
                    match ors.len() {
                        [] => (),
                        [or] => ands.push(or),
                        ors => ands.push(FilterCondition::Or(ors)),
```

Can't we do something like that instead?

Reply:
I tried to your suggestion in local and executed clippy, but compile errors occurred.

<img width="1247" alt="Screen Shot 2022-11-01 at 21 47 22" src="https://user-images.githubusercontent.com/38400669/199235711-3a42c5fd-8115-46fe-b82d-98b9b7ebb5d1.png">

type mismatch for `ors`.

Reply:
Indeed, as this is a `Vec` and not a slice it can't work this way but you can probably do that:
```suggestion
                    match ors.len() {
                        0 => (),
                        1 => ands.push(ors.pop().unwrap()),
                        _ => ands.push(FilterCondition::Or(ors)),
```

Reply:
Thanks for suggestion, I follow the code and it passed clippy run.

Reply:
Thanks for implementing the change @unvalley!!